src/__tests__/agent.test.ts

```diff
@@ -0,0 +1,59 @@
+import ServerMock = require("mock-http-server");
+import { getHttpFetch, getHttpsFetch } from "../agent";
+
+const TEST_HOST = "localhost";
+const TEST_PORT = 40001;
+
+// tslint:disable-next-line: no-any
+function createServerMock(): any {
+  const server = new ServerMock(
+    { host: TEST_HOST, port: TEST_PORT },
+    undefined
+  );
+
+  server.on({
+    method: "GET",
+    path: "/agent",
+    reply: {
+      status: 200
+    }
+  });
+
+  return server;
+}
+
+// tslint:disable-next-line: no-object-mutation
+const envWithKeepalive = {
+  FETCH_KEEPALIVE_ENABLED: "true"
+};
+
+describe("HttpAgentKeepAlive", () => {
```

Comment:
It would be nice to test the reuse of the TCP connection - it should be enough to check whether the source port is the same between two requests

Reply:
done